src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/move_readvalue_inputs_to_subgraph.cpp

```diff
@@ -0,0 +1,187 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "transformations/cpu_opset/common/op/read_value_with_subgraph.hpp"
+#include "transformations/cpu_opset/common/op/sdpa.hpp"
+#include "move_readvalue_inputs_to_subgraph.hpp"
+
+#include <unordered_set>
+
+#include "itt.hpp"
+#include "openvino/core/rt_info.hpp"
+#include "openvino/pass/constant_folding.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "ov_ops/rotary_positional_embeddings.hpp"
+#include "transformations/rt_info/disable_fp16_compression.hpp"
+#include "transformations/utils/gen_pattern.hpp"
+#include "transformations/utils/utils.hpp"
+#include "transformations/cpu_opset/common/op/submodel.hpp"
+
+// Check if skip pattern:
+//         ReadValue
+//             |
+//       Convert(Optional)
+//             |
+// ScaledDotProductAttentionWithKVCache
+//             |
+//       Convert(Optional)
+//             |
+//          Assign
+inline bool skip_pattern_sdpa(std::shared_ptr<ov::op::v6::ReadValue> readvalue) {
+    for (const auto& node : readvalue->get_output_target_inputs(0)) {
+        auto sdpa =
+            ov::as_type_ptr<ov::intel_cpu::ScaledDotProductAttentionWithKVCache>(node.get_node()->shared_from_this());
+        if (!sdpa) {
+            // Check whether ReadValue's grandson is SDPA.
+            for (const auto& child : node.get_node()->get_output_target_inputs(0)) {
+                sdpa = ov::as_type_ptr<ov::intel_cpu::ScaledDotProductAttentionWithKVCache>(
+                    child.get_node()->shared_from_this());
+                if (sdpa) {
+                    break;
+                }
+            }
+        }
+
+        if (sdpa) {
+            for (size_t port = 0; port < sdpa->get_output_size(); port++) {
+                for (const auto& child : sdpa->get_output_target_inputs(port)) {
+                    auto assign = ov::as_type_ptr<ov::op::v6::Assign>(child.get_node()->shared_from_this());
+                    if (!assign) {
+                        // Check whether SDPA's grandson is Assign.
+                        for (const auto& grandson : child.get_node()->get_output_target_inputs(0)) {
+                            assign = ov::as_type_ptr<ov::op::v6::Assign>(grandson.get_node()->shared_from_this());
+                            if (assign) {
+                                break;
+                            }
+                        }
+                    }
+                    if (assign) {
+                        if (assign->get_variable_id() == readvalue->get_variable_id()) {
+                            return true;
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Do we really need this check? Isn't it just enough to make sure that there is a custom SDPA node to make a decision to cancel the transformation?

Reply:
Yes, remove,  decision code to stateful_sdpa_fusion.cpp.